R/onevax_xvw_trial.R

```diff
@@ -97,8 +99,21 @@ vax_params_xvw_trial <- function(vea = 0, vei = 0, ved = 0, ves = 0,
   i_p <- i[i %% n_diag_rec != 1]
 
   # create diagnosis history mapping
-  diag_rec <- create_vax_map_branching(idx$n_vax, c(0, 1), i_eligible, i_p,
-                                       set_vbe = FALSE, idx)
+  #symptomatic diagnoses always recorded
+  diag_rec_s <- create_vax_map_branching(idx$n_vax, c(0, 1), i_eligible, i_p,
+                                         set_vbe = FALSE, idx)
+
+  #asymptomatic diagnoses may or may not be recorded
+  if (asymp_recorded == TRUE) {
+    diag_rec_a <- create_vax_map_branching(idx$n_vax, c(0, 1), i_eligible, i_p,
+                                           set_vbe = FALSE, idx)
+  } else if (asymp_recorded == FALSE) {
```

Comment:
```suggestion
  if (asymp_recorded) {
    diag_rec_a <- diag_rec_s
  } else {
```
Bad practice to write `if (x==TRUE)` always write `if(x)` instead